code/game/objects/items/weapons/storage/backpack.dm

```diff
@@ -18,6 +18,34 @@
 	var/side = 0 //0 = right, 1 = left
 	drop_sound = 'sound/items/drop/backpack.ogg'
 	pickup_sound = 'sound/items/pickup/backpack.ogg'
+	var/list/species_restricted = null
+
+/obj/item/storage/backpack/mob_can_equip(M, slot, disable_warning = FALSE)
+
+	//if we can't equip the item anyway, don't bother with species_restricted (cuts down on spam)
+	if (!..())
+		return 0
+
+	if(LAZYLEN(species_restricted) && istype(M,/mob/living/carbon/human))
+		var/exclusive = null
+		var/wearable = null
+		var/mob/living/carbon/human/H = M
+
+		if("exclude" in species_restricted)
+			exclusive = 1
+
+		if(H.species)
+			if(exclusive)
+				if(!(H.species.get_bodytype(H) in species_restricted))
+					wearable = 1
+			else
+				if(H.species.get_bodytype(H) in species_restricted)
+					wearable = 1

```

Comment:
```suggestion
			wearable = exclusive ^ (H.species.get_bodytype(H) in species_restricted)
```
XOR is true if an odd number of the inputs is true - either exclusive is true and bodytype is not in the list, or exclusive is false and bodytype _is_ in the list.